lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -682,3 +684,36 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
   removeCall();
   return true;
 }
+
+bool swift::ArraySemanticsCall::replaceByAppendingValues(SILModule &M, SILFunction* appendFn, llvm::SmallVectorImpl<SILValue> &Vs, ArrayRef<ProtocolConformanceRef> &conformances) {
```

Comment:
May be you should form the array of substitutions in the caller of this function and pass it as an argument, i.e. as `ArrayRef<Substitution>`. It would make the signature simpler and would make the code cleaner.